addons/hw_scale/controllers/main.py

```diff
@@ -158,11 +196,11 @@ def set_status(self, status, message=None):
             elif status == 'disconnected' and message:
                 _logger.info('Disconnected Scale: %s', message)
 
-    def _get_raw_response(self, connection):
+    def _get_raw_response(self, connection, protocol):
         answer = []
         while True:
             char = connection.read(1) # may return `bytes` or `str`
-            if not char:
+            if not char or protocol.outputTerminator in char:
```

Comment:
Shouldn't be '=', and check that there is no problem if outputTerminator = None (None in char)